src/server/metrics.rs

```diff
@@ -39,10 +40,23 @@ impl Metrics {
         StatsdClient::builder("", NopMetricSink).build()
     }
 
-    // TODO: Return this as a metric string for testing/debugging?
+    // increment a counter with no tags data.
     pub fn incr(self, label: &str) {
+        self.incr_with_tags(label, None)
+    }
+
+    pub fn incr_with_tags(self, label: &str, tags: Option<HashMap<String, String>>) {
         if self.client.is_some() {
-            match self.client.unwrap().incr(label) {
+            let client = self.client.unwrap();
+            let mut incr = client.incr_with_tags(label);
+            let tags = tags.unwrap_or_default();
+            let keys = tags.keys();
+            for tag in keys {
+                incr = incr.with_tag(tag, &tags.get(tag).unwrap())
+            }
+            // Include any "hard coded" tags.
+            incr = incr.with_tag("version", env!("CARGO_PKG_VERSION"));
```

Comment:
if we bump the minor/patch version, we probably aren't going to want to change the metrics dashboards. Would you mind also grabbing the `CARGO_PKG_VERSION_MAJOR` here too so it'll make for less required tracking/dashboard changes when version numbers change?

Reply:
I'm not sure passing along the version is necessary. I think we can distinguish between the old python cluster and new syncstorage-rs metrics easily: https://github.com/mozilla-services/syncstorage-rs/issues/222#issuecomment-540153554

Also our metrics tend to be configured to include tags for further distinguishing too, e.g. "host" and  "app" tags by the ingesting process (telegraf in syncstorage's case). We might wanna double check this w/ @bobm